Zapper-fi/studio
# feat: Dynamically infer app dependencies at boot, dynamically use externallyConfigured

## Description

- It's a dev UX painpoint to manually add dependency modules like `uniswap-v2` or `synthetix` when working with a app module that is a dependent of these helpers
- It's also painpoint to use `externallyConfigured` or `ExternalAppImport`. 
- Instead, expose a custom decorator `Register.AppModule` which adds `APP_ID` metadata to the class
- Use this metadata to derive dependencies in dev mode
- Also use this metadata to automatically update `imports` array to swap the module for its externally configured instance

## Checklist


## How to test?

Enable `pickle` only in `.env`, then when you start, you'll see:

<img width="982" alt="image" src="https://user-images.githubusercontent.com/22452366/165851156-77076083-37a5-4234-9020-b25ee986e16d.png">
